src/win/getaddrinfo.c

```diff
@@ -311,6 +320,14 @@ int uv_getaddrinfo(uv_loop_t* loop,
 
   /* save alloc_ptr now so we can free if error */
   req->alloc = (void*)alloc_ptr;
+  work = NULL;
```

Comment:
Do this before the first `goto error` statement.  It's potentially freeing a wild pointer now.

Reply:
Good catch.